### PR TITLE
fix(useRouteParams): deconstruct route to separate the path

### PR DESCRIPTION
**Pull request by Tzyito**

### CODE REVIEW

**Comment by Tzyito** [comment]
@anteriovieira thx for your hard work

**Comment by anteriovieira** [comment]
Thanks for you help @Tzyito

File: packages/router/useRouteParams/index.ts [hunk 1]
```diff
@@ -64,7 +64,9 @@ export function useRouteParams<
         trigger()
 
         nextTick(() => {
-          router[toValue(mode)]({ ...route, params: { ...route.params, ...Object.fromEntries(_params.entries()) } })
+          const { name: pathName, path, ...otherRouteParams } = route
+          const routeParams = { name: pathName, ...otherRouteParams }
+          router[toValue(mode)]({ ...routeParams, params: { ...route.params, ...Object.fromEntries(_params.entries()) } } as RouteLocationRaw)
```
Comment (anteriovieira) [inline packages/router/useRouteParams/index.ts]: Hi @Tzyito , We don't need the other props from the route. What do you think about using just what we need?

```ts
        nextTick(() => {
          const { params, query, hash } = route;

          router[toValue(mode)]({
            params: {
              ...params,
              ...Object.fromEntries(_params),
            },
            query,
            hash,
          });
        });
```

Comment (Tzyito) [reply packages/router/useRouteParams/index.ts]: @anteriovieira oh,sorry.i think you are right

Comment (anteriovieira) [reply packages/router/useRouteParams/index.ts]: no problem, I'll merge as soon as you update

**Pull request merged**